clients/roscpp/src/libros/rosout_appender.cpp

```diff
@@ -102,7 +103,14 @@ void ROSOutAppender::log(::ros::console::Level level, const char* str, const cha
   msg->file = file;
   msg->function = function;
   msg->line = line;
-  this_node::getAdvertisedTopics(msg->topics);
+  
+  // check parameter server/cache for omit_topics flag
+  // the same parameter is checked in rosout.py for the same purpose
+  ros::param::getCached("/rosout_disable_topics_generation", disable_topics_);
```

Comment:
this subscribes almost all roscpp-based nodes to param server updates

Reply:
`getCached` calls `ROS_DEBUG_NAMED` under the hood. Was this tested with debug log level=